## **Lab 10 - Random Forest Classifier**
### S Kailash 18MIS1074 
### SWE4012 Machine Learning (L11+L12)
### Faculty : Dr. M Premlatha
### Date : 12th April 2022

In this experiment, we will be doing Random forest algorithm using the sklearn ensemble module. We will be considering two datasets, the iris dataset and wine dataset. Let us look at the code now

In [27]:
import pandas as pd 
import numpy as np 
from sklearn.model_selection import train_test_split 
from sklearn.tree import DecisionTreeClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.metrics import accuracy_score
from sklearn.datasets import load_iris,load_wine
from sklearn import tree

In [19]:
df=load_iris()
x=df.data
y=df.target

In [20]:
X_train,X_test,y_train,y_test=train_test_split(x,y,random_state=50, test_size=0.25)

In [28]:
rfc=RandomForestClassifier()
rfc.fit(X_train,y_train)

RandomForestClassifier()

In [29]:
y_pred=dtc.predict(X_test)
print("Accuracy - ",accuracy_score(y_test,y_pred))

Accuracy -  0.9333333333333333


As we can see, we got an accuracy score of ~0.934 with the Random Forest Algorithm for Iris dataset. Let us see how it proceeds for our other dataset, the wine dataset.

In [23]:
df2=load_wine()
x=df2.data
y=df2.target

In [24]:
X_train,X_test,y_train,y_test=train_test_split(x,y,random_state=50, test_size=0.25)

In [30]:
rfc=RandomForestClassifier()
rfc.fit(X_train,y_train)

RandomForestClassifier()

In [31]:
y_pred=dtc.predict(X_test)
print("Accuracy - ",accuracy_score(y_test,y_pred))

Accuracy -  0.9333333333333333


We can see that it has an accuracy of 0.933, same as the previously achieved accuracy of 0.933. Therefore, the accuracy for prediction data is equally good on the iris dataset, as compared to the wine dataset.

Thank You!